helix-term/src/commands.rs

```diff
@@ -3993,6 +3949,226 @@ pub fn completion(cx: &mut Context) {
         },
     );
 }
+fn completion_word(cx: &mut Context) {
+    completion_word_impl(cx, true)
+}
+fn completion_word_impl(cx: &mut Context, search_root: bool) {
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+    let cursor = doc.selection(view.id).primary().cursor(text);
+    use helix_core::chars;
+    let mut iter = text.chars_at(cursor);
+    iter.reverse();
+    let offset = iter.take_while(|ch| chars::char_is_word(*ch)).count();
+    let start_offset = cursor.saturating_sub(offset);
+    let prefix = regex::escape(&text.slice(start_offset..cursor).to_string());
+
+    completion_grep(
+        cx,
+        format!(r"\b{}\w+", prefix,),
+        start_offset,
+        search_root,
+        false,
+    );
```

Comment:
Using a regex to extract words is a lot less efficient than just scanning the rope. Here's another approach that does that https://github.com/helix-editor/helix/pull/3328